python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -2157,3 +2171,73 @@ def inject_env_vars(self, location_name: Optional[str]):
         new_env = self._secrets_loader.get_secrets_for_environment(location_name)
         for k, v in new_env.items():
             os.environ[k] = v
+
+    def get_current_logical_version(
```

Comment:
It looks like most of this is business logic rather than storage access logic, and it's called in places that are already branching on `is_source`. Instead of adding this method on the instance, could the calling code just invoke a method that does the business logic? Or move it into `extract_logical_version_from_entry`?